input/script_utils.go

```diff
@@ -666,13 +754,22 @@ func senderHtlcTapScriptTree(senderHtlcKey, receiverHtlcKey,
 // The top level keyspend key is the revocation key, which allows a defender to
 // unilaterally spend the created output.
 func SenderHTLCScriptTaproot(senderHtlcKey, receiverHtlcKey,
-	revokeKey *btcec.PublicKey, payHash []byte) (*HtlcScriptTree, error) {
+	revokeKey *btcec.PublicKey, payHash []byte,
+	localCommit bool) (*HtlcScriptTree, error) {
+
+	var hType htlcType
+	if localCommit {
+		hType = htlcLocalOutgoing
+	} else {
+		hType = htlcRemoteIncoming
```

Comment:
I think this is `htlcRemoteOutgoing` since if we are spending an outgoing htlc via direct preimage spend, we wanna end up using `h.SuccessTapLeaf.Script`?

Reply:
Yeah this is pretty subtle:
  * This is the sender HTLC script. 
  * If we're the sender, then this is an outgoing HTLC on our local commitment. 
  * For the remote party, this is an incoming HTLC on their commitment. 

Otherwise we'll end up signing the wrong version and hit an HTLC mismatch. If you flip these, then try to run the channel state machine test `TestSimpleAddSettleWorkflow`, you'll see it fails. 

I'm not sure we can ever 100% eliminate the sort of ambiguity here re the local/remote terminology given the asymmetric state of the commitments. At least with this lates refactor, we can push down the call site that needs to handle it a bit, and the caller just says "give me the script to sign", "or the script to use for a revocation".